### PR TITLE
JSUI-3015 Added unit tests and accessibility tests for `SmartSnippet`

### PR DESCRIPTION
**Pull request by btaillon**
Along with those unit tests and accessibility tests, some minor tweaks were made.

For instance, closing a modal by pressing the send button no longer sends both the "reason" analytics and the "modal closed" analytics, it only sends the "reason".

Additionally, the following accessibility changes were made:
Screen Shot 2020-07-13 at 9 48 26 AM

Screen Shot 2020-07-13 at 9 50 39 AM

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: src/ui/SmartSnippet/ExplanationModal.ts [hunk 1]
```diff
@@ -68,26 +85,32 @@ export class ExplanationModal {
...
   private buildExplanations() {
-    const explanationsContainer = $$('div', { className: EXPLANATIONS_CLASSNAME }).el;
+    const explanationsContainer = $$('fieldset', { className: EXPLANATIONS_CLASSNAME }, this.buildExplanationsLabel()).el;
     this.reasons = this.options.explanations.map(explanation => this.buildExplanationRadioButton(explanation));
     this.reasons[0].select();
     this.reasons.forEach(radioButton => explanationsContainer.appendChild(radioButton.getElement()));
     return explanationsContainer;
   }
 
+  private buildExplanationsLabel() {
```
Comment (samisayegh) [inline src/ui/SmartSnippet/ExplanationModal.ts:95]: Could `buildReasonsLabel` be a more consistent name?

File: sass/_SmartSnippet.scss [hunk 2]
```diff
@@ -168,8 +168,8 @@
           }
 
           &.coveo-user-feedback-banner-yes-button {
-            color: $color-green;
-            fill: $color-green;
+            color: $color-strong-contrast-green;
```
Comment (ThibodeauJF) [inline sass/_SmartSnippet.scss:171]: it has a strong contrast on white, not necessarily all the time, would `$color-dark-green` be a better name?

File: strings/strings.json [hunk 3]
```diff
@@ -7596,5 +7596,9 @@
   "Send": {
     "en": "Send",
     "fr": "Envoyer"
+  },
+  "UsefulnessFeedbackReason": {
+    "en": "What's wrong?",
```
Comment (fbeaudoincoveo) [inline strings/strings.json:7601]: I'd go with:

Reason / Raison

as a legend

File: src/ui/SmartSnippet/ExplanationModal.ts (hunk 1 above)
Comment (btaillon) [reply src/ui/SmartSnippet/ExplanationModal.ts:95]: You're right, I'll rename it along with most mentions of `explanation`.

File: sass/_SmartSnippet.scss (hunk 2 above)
Comment (btaillon) [reply sass/_SmartSnippet.scss:171]: Good point, I'll rename it.

File: strings/strings.json (hunk 3 above)
Comment (btaillon) [reply strings/strings.json:7601]: Perfect, I'll change it. Thank you!

**Pull request merged**